.github/workflows/zip-templates-on-release.yml

```diff
@@ -0,0 +1,32 @@
+name: zip-templates-on-release
+on:
+  release:
+    types: [published]
+jobs:
+  zip-templates:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - run: npm install -g yarn
+      - run: yarn install
+      - run: sudo apt-get update && sudo apt-get install -y --allow-unauthenticated zip
+      - run: curl -s https://api.github.com/repos/ONSdigital/design-system/releases/latest | jq -r ".tarball_url" | wget -i - -O source.tar.gz
+      - run: mkdir design-system
+      - run: tar -xzf source.tar.gz -C design-system --strip-components=1
+      - run: cd design-system
+      - run: echo "GITHUB_RELEASE_TAG=$(curl -s https://api.github.com/repos/ONSdigital/design-system/releases/latest | jq -r ".tag_name")" >> $GITHUB_ENV
```

Comment:
You don't need to hit the GitHub API for the release version. This release event has multiple payload properties that can be used: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release
The `GITHUB_REF` env var will give you the tag.